spacepy/pybats/dgcpm.py

```diff
@@ -99,63 +103,76 @@ def __init__(self, filename, *args, **kwargs):
-                 ('vphi','m/s'), ('source','m^-3'),('fluxphi','m^{-3}s^{-1}'),
-                 ('fluxr','m^{-3}s^{-1}'), ('totaln','#'), ('vol','km^3')]
+        units = [('n', 'cm^{-3}'), ('x', 'R_E'), ('y', 'R_E'),
+                 ('b_open', 'boolean'), ('theta', 'degrees'),
+                 ('phi', 'degrees'), ('pot', 'kV',), ('corot', 'kV'),
+                 ('vr', 'm/s'), ('vphi', 'm/s'), ('source', 'm^-3'),
+                 ('fluxphi', 'm^{-3}s^{-1}'), ('fluxr', 'm^{-3}s^{-1}'),
+                 ('totaln', '#'), ('vol', 'km^3')]
         units = dict(units)
-        
+
+        # Open file; unzip if necessary.
+        if self.attrs['file'][-3:] == '.gz':
+            infile = gzip.open(self.attrs['file'], 'rt')
+        else:
+            infile = open(self.attrs['file'], 'r')
+
         # Read the header of the file.
-        infile = open(filename, 'r')
         parts = infile.readline().split()
         nLat, nLon = int(parts[-2]), int(parts[-1])
-        self.attrs['time']=dt.datetime.strptime(parts[0],'T=%Y%m%d_%H%M%S_000')
+        self.attrs['time'] = dt.datetime.strptime(parts[0],
+                                                  'T=%Y%m%d_%H%M%S_000')
         varlist = infile.readline().lower().split()[2:]
 
         # Create containers for data:
         for v in varlist:
-            self[v] = dmarray(np.zeros( (nLat,nLon) ), {'units':units[v]})
+            unit = '' if v not in units else units[v]
+            self[v] = dmarray(np.zeros((nLat, nLon)), {'units': unit})
 
         # Read rest of file and sort data into arrays:
-        for l in infile.readlines():
-            parts = l.split()
-            i,j = int(parts.pop(0))-1, int(parts.pop(0))-1
+        for line in infile.readlines():
+            parts = line.split()
+            i, j = int(parts.pop(0))-1, int(parts.pop(0))-1
             for v, x in zip(varlist, parts):
-                self[v][i,j] = x
+                self[v][i, j] = x
 
         # That's it for our file.
         infile.close()
 
         # Create some "helper" variables:
-        self['lat']=dmarray(self['theta'][:,0], {'units':'degrees'})
-        self['lon']=dmarray(self['phi'][  0,:], {'units':'degrees'})
-        self['mlt']=dmarray(self['lon']/15.0,   {'units':'hours'})
+        self['lat'] = dmarray(self['theta'][:, 0], {'units':'degrees'})
+        self['lon'] = dmarray(self['phi'][0, :], {'units':'degrees'})
+        self['mlt'] = dmarray(self['lon']/15.0, {'units':'hours'})
 
         # Calculate L-Shell for convenience.
-        self['L'] = dmarray(np.cos(self['lat']*np.pi/180.)**-2,{'units':'R_E'})
+        self['L'] = dmarray(np.cos(self['lat']*np.pi/180.)**-2,
+                            {'units':'R_E'})
 
         # Sometimes, SI units aren't the best.
-        if 'n'     in self: self['n']    /= 100**3
-        if 'pot'   in self: self['pot']  /= 1000.0
-        if 'corot' in self: self['corot']/= 1000.0
+        if 'n' in self:
+            self['n'] /= 100**3
```

Comment:
Probably should be 100.**3 ... if self['n'] is an integer numpy array (yes, that would be weird) this would fail.